src/orquestra/integrations/qiskit/backend/backend.py

```diff
@@ -374,15 +377,19 @@ def _apply_readout_correction(
             num_qubits = len(key)
             break
 
-        if active_qubits is None:
-            active_qubits = list(range(num_qubits))
+        if virtual_to_physical_qubits_dict is None:
+            virtual_qubits = list(range(num_qubits))
+            physical_qubits = list(range(num_qubits))
         else:
-            active_qubits.sort()
+            virtual_qubits = list(virtual_to_physical_qubits_dict.keys())
+            physical_qubits = list(virtual_to_physical_qubits_dict.values())
+            virtual_qubits.sort()
+            physical_qubits.sort()
             for key in deepcopy(list(counts.keys())):
```

Comment:
You may consider using `qiskit.result.marginal_counts` instead of the loop (https://qiskit.org/documentation/stubs/qiskit.result.marginal_counts.html)

Reply:
Sounds good. Will try it.